packages/salesforcedx-vscode-soql/src/editor/soqlEditorProvider.ts

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import * as fs from 'fs';
+import * as path from 'path';
+import * as vscode from 'vscode';
+
+export class SOQLEditorProvider implements vscode.CustomTextEditorProvider {
+  public static register(context: vscode.ExtensionContext) {
+    const provider = new SOQLEditorProvider(context);
+    const providerRegistration = vscode.window.registerCustomEditorProvider(
+      SOQLEditorProvider.viewType,
+      provider
+    );
+    return providerRegistration;
+  }
+
+  private static readonly viewType = 'soqlCustom.soql';
+  constructor(private readonly context: vscode.ExtensionContext) {}
+
+  public async resolveCustomTextEditor(
+    document: vscode.TextDocument,
+    webviewPanel: vscode.WebviewPanel,
+    _token: vscode.CancellationToken
+  ): Promise<void> {
+    webviewPanel.webview.options = {
+      enableScripts: true,
+      localResourceRoots: [
+        vscode.Uri.file(path.join(this.context.extensionPath, 'node_modules'))
+      ]
+    };
+
+    webviewPanel.webview.html = this.getWebViewContent(webviewPanel.webview);
+  }
+
+  private getWebViewContent(webview: vscode.Webview): string {
+    const pathToLwcDist = path.join(
+      this.context.extensionPath,
+      'node_modules/@salesforce/soql-builder-ui/dist'
+    );
+    const pathToHtml = path.join(pathToLwcDist, 'index.html');
+    let html = fs.readFileSync(pathToHtml).toString();
+    const scriptUri = webview.asWebviewUri(
+      vscode.Uri.file(path.join(pathToLwcDist, 'app.js'))
+    );
+    const zeroDotScriptUri = webview.asWebviewUri(
+      vscode.Uri.file(path.join(pathToLwcDist, '0.app.js'))
+    );
+
+    html = html.replace('./0.app.js', `${zeroDotScriptUri}`);
+    html = html.replace('./app.js', `${scriptUri}`);
```

Comment:
Need to replace the HTML `src` attributes with URIs that vs code can use in the webview.